cabal-install/Distribution/Client/ProjectConfig.hs

```diff
@@ -614,9 +614,14 @@ renderBadPackageLocations (BadPackageLocations prov bpls)
 -- informative help text about the issues related to having no project
 -- file and no package.
 renderImplicitBadPackageLocation :: BadPackageLocation -> String
-renderImplicitBadPackageLocation bpl =
-    case bpl of
-        _ -> renderBadPackageLocation bpl
+renderImplicitBadPackageLocation bpl = case bpl of
+    BadLocGlobEmptyMatch pkglocstr ->
+        "No cabal.project or cabal file matching the glob '"
+     ++ pkglocstr ++ "' was found.\n"
+     ++ "Please create a package description file <pkgname>.cabal "
+     ++ "or a cabal.project file referencing the packages you  "
+     ++ "want to build."
+    _ -> renderBadPackageLocation bpl
```

Comment:
Regarding the other cases, I'm not convinced I actually understand well enough under what situations these errors are raised to correctly word the implicit error message.